src/Nirum/Targets/Python.hs

```diff
@@ -1069,69 +1055,235 @@ compileTypeDeclaration src d@ServiceDeclaration { serviceName = name'
         raise NotImplementedError(
-            '#{className} has to implement #{toAttributeName' mName}()'
+            '#{className} has to implement #{toAttributeName' (methodName m)}()'
         )
+
+    #{toAttributeName' (methodName m)}.__nirum_argument_serializers__ = {
+    }  # type: typing.Mapping[str, typing.Callable[[object], object]]
+    #{toAttributeName' (methodName m)}.__nirum_argument_deserializers__ = {
+    }
+%{ forall (Parameter pName pt _, tx, (Validator pTPred pValVs), d) <- params' }
+    def __nirum_argument_serializer__(#{toAttributeName' pName}):
+        if not (#{pTPred}):
+            raise #{builtins}.TypeError(
+                '#{toAttributeName' pName} must be a value of ' +
+                #{typing}._type_repr(#{tx}) + ', not ' +
+                #{builtins}.repr(#{toAttributeName' pName})
+            )
+%{ forall ValueValidator pValuePredCode pValueErrorMsg <- pValVs }
+        elif not (#{pValuePredCode}):
+            raise #{builtins}.ValueError(
+                'invalid #{toAttributeName' pName}: '
+                #{stringLiteral pValueErrorMsg}
+            )
+%{ endforall }
+        return #{compileSerializer' src pt (toAttributeName' pName)}
+    __nirum_argument_serializer__.__name__ = '#{toAttributeName' pName}'
+    #{toAttributeName' (methodName m)}.__nirum_argument_serializers__[
+        '#{toAttributeName' pName}'] = __nirum_argument_serializer__
+    del __nirum_argument_serializer__
+
+    def __nirum_argument_deserializer__(value, on_error=None):
+        on_error = #{defaultErrorHandler}(on_error)
+#{indent "        " d}
+        on_error.raise_error()
+        if not on_error.errored:
+            return rv
+    __nirum_argument_deserializer__.__name__ = '#{toBehindSnakeCaseText pName}'
+    #{toAttributeName' (methodName m)}.__nirum_argument_deserializers__[
+        '#{toBehindSnakeCaseText pName}'] = __nirum_argument_deserializer__
+    del __nirum_argument_deserializer__
+%{ endforall }
+
+    def __nirum_serialize_arguments__(
+%{ case pyVer }
+%{ of Python3 }
+%{ forall (Parameter pName _ _, pTypeExpr, _, _) <- params' }
+        #{toAttributeName' pName}: '#{pTypeExpr}',
+%{ endforall }
+    ) -> '#{rTypeExpr}':
+%{ of Python2 }
+%{ forall (Parameter pName _ _, _, _, _) <- params' }
+        #{toAttributeName' pName},
+%{ endforall }
+    ):
+%{ endcase }
+        return {
+%{ forall (Parameter (Name fn bn) _ _, _, _, _) <- params' }
+            '#{I.toSnakeCaseText bn}':
+                #{className}.#{toAttributeName' (methodName m)}
+                    .__nirum_argument_serializers__['#{toAttributeName fn}'](
+                    #{toAttributeName fn}
+                ),
+%{ endforall }
+        }
+    #{toAttributeName' (methodName m)}.__nirum_serialize_arguments__ = \
+        __nirum_serialize_arguments__
+    del __nirum_serialize_arguments__
+
+    def __nirum_deserialize_arguments__(value, on_error=None):
```

Comment:
Since `__nirum_deserialize_arguments__` takes mapping data, we can check a type value which one should be `Mapping` type.